version: "3"

services:
  mediawiki:
    environment: 
      WIKI_DEBUG: "true"
    labels: 
      traefik.http.routers.wiki-secure.tls.certresolver: pebble
  reverse-proxy:
    environment:
      TRAEFIK_LOG_LEVEL: DEBUG
      TRAEFIK_API_DASHBOARD: "true"
      TRAEFIK_API_INSECURE: "true"
      LEGO_CA_CERTIFICATES: /var/pebble-test-certs/pebble.minica.pem
      TRAEFIK_CERTIFICATESRESOLVERS_pebble_ACME_TLSCHALLENGE: "true"
      TRAEFIK_CERTIFICATESRESOLVERS_pebble_ACME_EMAIL: ${ACME_EMAIL}
      TRAEFIK_CERTIFICATESRESOLVERS_pebble_ACME_CASERVER: https://pebble:14000/dir
    volumes:
      - pebble-test-certs:/var/pebble-test-certs
    ports:
      - 8080:8080
    networks:
      - acme

  acme-pebble:
    image: letsencrypt/pebble
    # when container stop all account in memory will lost
    # that cause error to acme client which save account crendential already,
    # so we remove all acme data each start
    command: 
      - sh
      - -c 
      - | 
        rm -rf /var/acme/* &&
        pebble -config /test/my-pebble-config.json
    volumes:
      - acme-data:/var/acme
      - pebble-test-certs:/test/certs
      - ./pebble/my-pebble-config.json:/test/my-pebble-config.json
    networks:
      acme:
        aliases:
          - pebble
    extra_hosts:
      - ${WIKI_HOST}:host-gateway

volumes:
  pebble-test-certs:
networks:
  acme:
